src/app/routes/utils/sharedDataTransformers/addIndexesToEmbeds/index.js

```diff
@@ -20,16 +20,27 @@ const getEmbedProvider = getOembedProp('provider_name');
 const getEmbedIndexOfType = getOembedProp('indexOfType');
 const matchesEmbedProvider = provider =>
   pipe(getEmbedProvider, equals(provider));
-const enrichBlocks = (accumulator, block) => {
+const enrichBlocks = (accumulator, block, index, blocks) => {
   const embedUrl = getEmbedUrl(block);
   const embedProvider = getEmbedProvider(block);
 
   if (embedUrl && embedProvider) {
-    const blocksByProvider = accumulator.filter(
+    const accumulatedBlocksByProvider = accumulator.filter(
       matchesEmbedProvider(embedProvider),
     );
-    const numBlocksByProvider = blocksByProvider.length;
-    const lastBlockByProvider = blocksByProvider[numBlocksByProvider - 1];
+    const allBlocksByProvider = blocks.filter(
+      matchesEmbedProvider(embedProvider),
+    );
+    const numOfAccumulatedBlocksByProvider = accumulatedBlocksByProvider.length;
+    const numOfAllBlocksByProvider = allBlocksByProvider.length;
+    const lastBlockByProvider =
```

Comment:
can you pop this line to the line above line 44? just because `lastBlockByProvider` won't ever be used if `isOnlyEmbedOfType` is true. Thanks!